### PR TITLE
Accept long options, and add exclusive as per linux version

### PR DESCRIPTION
**Pull request by amaterasu-**
Not sure if you think this is valid for merge, but this pull request makes the utility command line compatible with the standard linux flock. Should be no functional change to operation.

### CODE REVIEW

File: src/flock.c [hunk 1]
```diff
@@ -165,7 +197,7 @@ int main(int argc, char *argv[]) {
 			timer.it_value.tv_sec = (time_t) raw_timeval;
 			timer.it_value.tv_usec = (suseconds_t) ((raw_timeval - timer.it_value.tv_sec) * 1000000);
 			break;
-		case '?':
+		case 'h':
```
Comment (josephholsten) [inline src/flock.c]: What does this do with unexpected flags?

Comment (amaterasu-) [inline src/flock.c]: Nothing good I'm afraid. I've reinstated the accidentally removed ? case.

**Comment by amaterasu-** [comment]
I think adding the long form options to tests (really duplicating the current tests, with different command lines) is the right call. I'll get that sorted today.

As to the posix compatibility, I was initially trying to only to -e and -x, but the build system I was working with was trying to use long options as well, so I would up adding the whole lot, so ideological purity was first port of call, until it didn't work :)

**Comment by amaterasu-** [comment]
Good call on the tests - I duffed the -w optarg and the handling of -c. Pull request is update to include testing on the long form opts (excluding the --close and -o case, I couldn't think of a good way of testing them either...)

**Comment by josephholsten** [comment]
LGTM, let grab it and make sure the tests pass.

I should really throw together a Travis config for this, but it still wouldn't really guarantee cross platform support.

**Comment by josephholsten** [comment]
@amaterasu- shucks, we didn't add the long opts and such to the documentation. I'm going to make sure it gets doc'd in #9.

**Pull request merged**